src/dsl/expr.rs

```diff
@@ -250,6 +320,41 @@ impl Operator {
     }
 }
 
+// Helper method that standardizes the error messages when "unary" expressions fail during type-resolution against a schema
+fn unary_resolving_type_error(
```

Comment:
check our thiserror crate instead for defining templates

Reply:
I ended up not using `thiserror`, instead just adding a new DaftError variant with some custom `Display::fmt` logic. (See PR summary for what the error looks like now for the user)

Reply:
sounds good, but im a bit confused on where we should return a TypeError vs the ExprResolveError. Can you write a note in the Error enum for it

Reply:
Ok added a note:

>     // ExprResolveTypeError: Typing error when resolving expressions against schemas
>     //
>     // This variant has custom Display logic for presenting a more user-friendly error message which shows
>     // exactly which operation, arguments and dtypes of those arguments caused the issue.